lib/curl/curl.nit

```diff
@@ -246,6 +249,116 @@ class CurlHTTPRequest
 	end
 end
 
+
+# Restful JSON request builder over HTTP
+#
+class CurlRestRequest
+	super CurlHTTPRequest
+	var method : nullable String
+	var jsonSrc : nullable Serializable is writable
+
+	redef fun execute : CurlResponse
+	do
+		# BASIC CURL STUFF (Reveiver, callback, user agent, URL, Address family)
+
+		var success_response = new CurlRestResponseSuccess
+
+		if not self.curl.is_ok then return answer_failure(0, "Curl instance is not correctly initialized")
+
+		var callback_receiver : CurlCallbacks = success_response
+		if self.delegate != null then callback_receiver = self.delegate.as(not null)
+
+		var err : CURLCode
+
+		err = self.curl.native.easy_setopt(new CURLOption.follow_location, 1)
+		if not err.is_ok then return answer_failure(err.to_i, err.to_s)
+
+		err = self.curl.native.easy_setopt(new CURLOption.url, url)
+		if not err.is_ok then return answer_failure(err.to_i, err.to_s)
+
+		var user_agent = user_agent
+		if user_agent != null then
+			err = curl.native.easy_setopt(new CURLOption.user_agent, user_agent)
+			if not err.is_ok then return answer_failure(err.to_i, err.to_s)
+		end
+
+		var unix_socket_path = unix_socket_path
+		if unix_socket_path != null then
+			err = self.curl.native.easy_setopt(new CURLOption.unix_socket_path, unix_socket_path)
+			if not err.is_ok then return answer_failure(err.to_i, err.to_s)
+		end
+
+		# CALLBACKS
+		err = self.curl.native.register_callback_header(callback_receiver)
+		if not err.is_ok then return answer_failure(err.to_i, err.to_s)
+
+		err = self.curl.native.register_callback_body(callback_receiver)
+		if not err.is_ok then return answer_failure(err.to_i, err.to_s)
+
+		# SET REQUEST HEADER
+		var headers = self.headers
+		if headers != null then
+			var headers_joined = headers.join_pairs(": ")
+			err = self.curl.native.easy_setopt(new CURLOption.httpheader, headers_joined.to_curlslist)
+			if not err.is_ok then return answer_failure(err.to_i, err.to_s)
+		end
+
+		# SET HTTP METHOD AND BODY
+		if self.method=="GET" then
+			err=self.curl.native.easy_setopt(new CURLOption.get, 1)
+
+		else if self.method=="POST" then
+			err=self.curl.native.easy_setopt(new CURLOption.post, 1)
+			if not err.is_ok then return answer_failure(err.to_i, err.to_s)
+			err=setBody
+
+		else if self.method=="DELETE" then
+			err=self.curl.native.easy_setopt(new CURLOption.custom_request, "DELETE")
+
+		else if self.method=="PUT" then
+			err=self.curl.native.easy_setopt(new CURLOption.custom_request,"PUT")
+			if not err.is_ok then return answer_failure(err.to_i, err.to_s)
+			err=setBody
+
+		else if self.method=="PATCH" then
+			err=self.curl.native.easy_setopt(new CURLOption.custom_request,"PATCH")
+			if not err.is_ok then return answer_failure(err.to_i, err.to_s)
+			err=setBody
+		else
+			print "METHOD "+self.method.to_s+" NOT IMPLEMENTED"
```

Comment:
factorize all the "CURLOption.custom_request" in the else branch and gives the the method as is.